source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -225,9 +225,7 @@ Http::FilterDataStatus Filter::decodeData(Buffer::Instance& data, bool end_strea
   }
 
   auto& hashing_util = Envoy::Common::Crypto::UtilitySingleton::get();
-  if (!decoder_callbacks_->decodingBuffer()) {
-    decoder_callbacks_->addDecodedData(data, false);
-  }
+  decoder_callbacks_->addDecodedData(data, false);
```

Comment:
Can you also cover this in a unit test and make sure you have the right mock expectations? This should definitely cause a UT failure.

Reply:
OK. The integration test fails without this change (as you expected).
I'll add the expectation in the unit tests.